src/Workleap.AspNetCore.Authentication.ClientCredentialsGrant/AuthenticationBuilderExtensions.cs

```diff
@@ -39,16 +39,17 @@ public static AuthenticationBuilder AddClientCredentials(this AuthenticationBuil
 
         builder.AddJwtBearer(authScheme, configureOptions);
 
-        var tokenHandlerNamedConfigureOptionsAlreadyAdded = builder.Services.Any(x => IsDescriptorOfJwtBearerOptionsValidator(x, authScheme));
-        if (!tokenHandlerNamedConfigureOptionsAlreadyAdded)
+        var areNamedSingletonsAlreadyRegistered = builder.Services.Any(x => IsNamedJwtBearerOptionsValidatorAlreadyRegistered(x, authScheme));
```

Comment:
I renamed this because I had to think twice about why we wrote this code. I hope it's more clear now.